frontend/public/components/hpa.tsx

```diff
@@ -34,64 +38,66 @@ const MetricsRow: React.FC<MetricsRowProps> = ({ type, current, target }) => (
 
 const externalRow = (metric, current, key) => {
   const { external } = metric;
-  const type = external.metricName;
+  const type = external.metric.name;
   // TODO: show metric selector for external metrics?
-  const currentValue = external.targetAverageValue
-    ? _.get(current, 'object.currentAverageValue')
-    : _.get(current, 'object.currentValue');
-  const targetValue = external.targetAverageValue
-    ? external.targetAverageValue
-    : external.targetValue;
+  const currentValue = current?.external?.current?.averageValue
```

Comment:
@spadgett I think line 43 should be `const currentValue = current?.external?.current?.averageValue` instead of `const currentValue = external.target.averageValue` since the conditional check is for `current?.external?.current?.averageValue`.